rhino/src/main/java/org/mozilla/javascript/NativeJSON.java

```diff
@@ -557,11 +565,20 @@ private static String quote(String string) {
                     }
                     break;
             }
+            prev = c;
         }
         product.append('"');
         return product.toString();
     }
 
+    static boolean isLeadingSurrogate(char c) {
```

Comment:
Are these checks specific to JSON or more general? If they are not specific to JSON i vote for haing a more central place for these methods. Not sure if we already have one (sorry i only have my cell phone at hand at the moment)

Reply:
No they aren't. Not sure where they should go though...

Reply:
@gbrail @p-bakker @tonygermano any suggestion?